### PR TITLE
doc: add summary of config changes and upgrading guides

### PR DESCRIPTION
**Pull request by junjiemao1**
This patch adds a summary of the significant changes to ACRN configuration and build in v2.4 and gives some instructions on how to upgrade from a previous release.

Signed-off-by: Junjie Mao <junjie.mao@intel.com>

### CODE REVIEW

**Review by fitchbe — APPROVED** [review]
Pushed copy edits.

File: doc/release_notes/release_notes_2.4.rst [hunk 1]
```diff
@@ -38,9 +38,124 @@ What's New in v2.4
...
+****************************************
+
+Users are highly recommended to follow the instructions below in order to
+upgrade to v2.4 from previous ACRN releases.
+
+Additional Dependencies
+=======================
+
+Python 3.6 or above is required to build ACRN v2.4. If you are using an old
+version of Ubuntu (e.g. 16.04), you can use PPA (e.g. `New Python Versions
```
Comment (dbkinder) [inline doc/release_notes/release_notes_2.4.rst:73]: I don't think we should recommend using unofficial releases of Python via deadsnakes.  Instead, let's ask them to check their python version (using ``python --version``) and if it's not version 3.6 or higher, use ``apt`` to check for an update.

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/release_notes/release_notes_2.4.rst (hunk 1 above)
Comment (junjiemao1) [reply doc/release_notes/release_notes_2.4.rst:73]: Unfortunately Ubuntu 16.04 only ships Python 3.5.2 in the official package repo.

**Comment by junjiemao1** [comment]
Added another section `New Configuration Options` which lists the added options in v2.4 and provides a script to upgrade a v2.3-compliant scenario XML.

File: doc/release_notes/release_notes_2.4.rst [hunk 2]
```diff
@@ -38,9 +38,160 @@ What's New in v2.4
...
+<https://launchpad.net/~deadsnakes/+archive/ubuntu/ppa>`_ from deadsnakes
+team) for recent Python versions.
+
+In addition, the following tools and packages are needed to build ACRN:
```
Comment (gvancuts) [inline doc/release_notes/release_notes_2.4.rst:77]: Can we make it clear that this is not the complete list but rather new tools that need to be installed compared to v2.3

I also ran into an issue whereby I had to install `libxml2-utils` which is currently not mentioned in our documentation (see PR #5862). This should also be mentioned here.

File: doc/release_notes/release_notes_2.4.rst [hunk 3]
```diff
@@ -38,9 +38,160 @@ What's New in v2.4
...
+     $ sudo pip3 install lxml xmlschema
+
+Configuration File Format
+=========================
+
+Starting from v2.4, Kconfig is no longer used, and the contents of scenario
+XMLs have been simplified. You need to upgrade your own Kconfig-format files
+or scenario XMLs if you maintain any.
+
+For Kconfig-format file, you must translate your configurations to a scenario
```
Comment (gvancuts) [inline doc/release_notes/release_notes_2.4.rst:91]:
```suggestion
For Kconfig-format file, you must translate your configuration to a scenario
```
?

**Review by gvancuts — APPROVED** [review]
Just some minor comments but otherwise LGTM!

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/release_notes/release_notes_2.4.rst (hunk 1 above)
Comment (junjiemao1) [reply doc/release_notes/release_notes_2.4.rst]: Thanks for the detailed instructions. I've updated the doc accordingly.

File: doc/release_notes/release_notes_2.4.rst (hunk 3 above)
Comment (junjiemao1) [reply doc/release_notes/release_notes_2.4.rst]: Typo fixed.

File: doc/release_notes/release_notes_2.4.rst (hunk 2 above)
Comment (junjiemao1) [reply doc/release_notes/release_notes_2.4.rst]: Thanks for pointing this out. I've added a note to call out that this is not the complete list. Also added libxml2-utils here.

**Review by gvancuts — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
LGTM, thanks!

**Pull request merged**